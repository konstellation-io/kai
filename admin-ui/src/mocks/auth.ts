import { GET_USER_EMAIL } from '../components/Header/Header.graphql';

export const usernameMock = {
  request: {
    query: GET_USER_EMAIL
  },
  result: {
    data: {
      me: {
        email: 'user@konstellation.com'
      }
    }
  }
};

export const unauthorizedUsernameMock = {
  request: {
    query: GET_USER_EMAIL
  },
  result: {
    data: {
      me: {
        email: 'unauthorizedUser@konstellation.com'
      }
    }
  },
  error: new Error('Unauthorized user')
};
