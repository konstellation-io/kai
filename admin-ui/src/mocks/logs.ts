import { GET_LOGS } from '../graphql/client/queries/getLogs.graphql';

export const currentLogPanelMock = {
  request: {
    query: GET_LOGS
  },
  result: {
    data: {
      runtimeId: 'runtimeIdMock',
      nodeId: 'nodeIdMock',
      nodeName: 'nodeNameMock'
    }
  }
};
