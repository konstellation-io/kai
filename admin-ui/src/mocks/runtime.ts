import { GET_DASHBOARD } from '../pages/Dashboard/Dashboard.graphql';
import { ADD_RUNTIME } from '../pages/AddRuntime/AddRuntime.graphql';

export const dashboardMock = {
  request: {
    query: GET_DASHBOARD
  },
  result: {
    data: {
      dashboard: {
        runtimes: [
          {
            id: '00001',
            status: 'ACTIVE',
            name: 'Some Name',
            creationDate: '2018-01-02'
          },
          {
            id: '00002',
            status: 'ACTIVE',
            name: 'Some Other Name',
            creationDate: '2018-02-03'
          }
        ],
        alerts: [
          {
            id: 'id0',
            type: 'error',
            message: 'some message',
            runtime: { id: '00001' }
          },
          {
            id: 'id1',
            type: 'error',
            message: 'some message 2',
            runtime: { id: '00001' }
          }
        ]
      }
    }
  }
};

export const addRuntimeMock = {
  request: {
    query: ADD_RUNTIME,
    variables: { name: 'New Runtime' }
  },
  result: {
    data: {
      createRuntime: {
        success: true,
        message: 'some message'
      }
    }
  }
};
