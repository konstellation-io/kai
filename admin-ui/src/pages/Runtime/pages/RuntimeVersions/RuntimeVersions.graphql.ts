import gql from 'graphql-tag';
import { Version } from '../../../../graphql/models';

export interface RuntimeVersionsResponse {
  versions: Version[];
}

export interface RuntimeVersionsVars {
  runtimeId?: string;
}

export const GET_VERSIONS = gql`
  query GetVersions($runtimeId: ID!) {
    versions(runtimeId: $runtimeId) {
      id
      name
      description
      status
      creationDate
      creationAuthor {
        email
      }
      activationDate
      activationAuthor {
        email
      }
      configurationCompleted
    }
  }
`;
