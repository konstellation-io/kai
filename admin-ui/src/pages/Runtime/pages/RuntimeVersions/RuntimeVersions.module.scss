@import '../../../../styles/mixins';
@import '../../../../styles/colors';

$color-active-bg: palette(gray, 750);

.container {
  height: 100%;
  display: flex;
  flex-direction: row;
}

.content {
  height: 100%;
  display: flex;
  flex-direction: column;
  padding: 40px 46px 32px 46px;
}

.subtitle {
  @include font-body;
  color: font-color(light);

  margin-bottom: 60px;
}

.activeVersion {
  @include border-header;

  margin-bottom: 32px;
  padding: 8px 24px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  background-color: $color-active-bg;
}

.versionTitle,
.versionName {
  @include font-body;
  color: font-color(light);
}

.versionTitle {
  line-height: 40px;
}

.versionCreation {
  @include font-caption;
  color: font-color(light);

  margin: 5px 25px;
}

.versionStatus {
  display: flex;
  flex-direction: row;
  align-items: center;
  margin: 25px auto 25px 29px;
}
.versionName {
  margin-right: 25px;
}

.greenCircle {
  width: 8px;
  height: 8px;
  margin-right: 10px;
  border-radius: 50%;
  background-color: $color-ok;
  flex-shrink: 0;
}

.secondRow {
  display: flex;
  flex-direction: row;
  flex-grow: 1;
  align-items: center;
  padding-right: 16px;
}

.versionList {
  @include listItems;

  display: relative;
  padding: 0 0 30px 0;
  overflow: overlay;
}
