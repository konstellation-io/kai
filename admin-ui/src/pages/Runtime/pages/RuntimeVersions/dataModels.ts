import gql from 'graphql-tag';

interface Version {
  id: number;
  model: string;
  year: number;
  stock: number;
}

export const GET_VERSIONS = gql`
  query GetVersions {
    versions {
      versionNumber
      description
      status
      creationDate
      creatorName
      activationDate
      activatorName
    }
  }
`;
