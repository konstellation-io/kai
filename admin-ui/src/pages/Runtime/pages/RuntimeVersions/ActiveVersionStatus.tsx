import React from 'react';

import Button from '../../../../components/Button/Button';
import { formatDate } from '../../../../utils/format';

import Check from '@material-ui/icons/Check';
import ErrorOutline from '@material-ui/icons/ErrorOutline';

import { RuntimeVersion } from './dataModels';

import cx from 'classnames';
import styles from './RuntimeVersions.module.scss';

type Props = {
  activeVersion: RuntimeVersion;
  onClick: Function;
};
function ActiveVersionStatus({ activeVersion, onClick }: Props) {
  const isVersionActive = activeVersion !== undefined;
  const title = isVersionActive
    ? 'Version active'
    : 'There is no active version';
  const Icon = isVersionActive ? Check : ErrorOutline;

  return (
    <div
      className={cx(styles.activeVersion, {
        [styles['active']]: isVersionActive
      })}
    >
      <Icon style={{ fontSize: '1rem' }} />
      <span className={styles.versionTitle}>{title}</span>
      {isVersionActive && (
        <>
          <span className={styles.versionCreation}>
            {`CREATED: ${formatDate(new Date(activeVersion.creationDate))}`}
          </span>
          <div className={styles.secondRow}>
            <div className={styles.versionStatus}>
              <div className={styles.greenCircle} />
              <span className={styles.versionName}>{'Version name'}</span>
            </div>
            <Button label="LOCATE THIS VERSION" onClick={onClick} border />
          </div>
        </>
      )}
    </div>
  );
}

export default ActiveVersionStatus;
