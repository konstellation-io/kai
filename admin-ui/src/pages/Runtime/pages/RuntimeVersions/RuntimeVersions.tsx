import { get } from 'lodash';

import React, { useRef } from 'react';
import { useParams } from 'react-router-dom';
import * as ROUTE from '../../../../constants/routes';

import Modal from '../../../../components/Modal/Modal';
import SpinnerCircular from '../../../../components/LoadingComponents/SpinnerCircular/SpinnerCircular';
import ErrorMessage from '../../../../components/ErrorMessage/ErrorMessage';
import ActiveVersionStatus from './ActiveVersionStatus';
import VersionInfo from '../../../../components/VersionInfo/VersionInfo';

import { useQuery } from '@apollo/react-hooks';
import {
  GET_VERSIONS,
  RuntimeVersionsResponse,
  RuntimeVersionsVars
} from './RuntimeVersions.graphql';
import { Version } from '../../../../graphql/models';

import styles from './RuntimeVersions.module.scss';

function RuntimeVersions() {
  const versionListRef = useRef(null);
  const { runtimeId } = useParams();
  const { data, loading, error } = useQuery<
    RuntimeVersionsResponse,
    RuntimeVersionsVars
  >(GET_VERSIONS, {
    variables: { runtimeId },
    fetchPolicy: 'no-cache'
  });

  if (loading) return <SpinnerCircular />;
  if (error) return <ErrorMessage />;

  let activeVersion: Version | undefined;
  if (data && data.versions.length !== 0) {
    activeVersion = data.versions.find(version => version.status === 'ACTIVE');
  }

  function onLocateActiveVersionClick() {
    const activeVersionInfoElement = document.getElementById(
      `versionInfoElement_${activeVersion &&
        activeVersion.description.replace(' ', '')}`
    );

    let scrollAmount = 0;

    const targetTop = activeVersionInfoElement
      ? activeVersionInfoElement.offsetTop
      : 0;

    const listTop: number = get(versionListRef, 'current.offsetTop') || 0;
    scrollAmount = targetTop - listTop;

    // @ts-ignore
    versionListRef.current.scrollTo({
      top: scrollAmount,
      behavior: 'smooth'
    });
  }

  function getContent() {
    if (data && data.versions.length === 0) {
      return (
        <Modal
          title="THERE IS NO RUNTIME VERSIONS"
          message="Please, upload a new version to start working on this runtime"
          actionButtonLabel="NEW VERSION"
          to={ROUTE.NEW_VERSION}
        />
      );
    }

    const versions =
      data &&
      data.versions.map((version: Version, idx: number) => (
        <VersionInfo key={`version_${idx}`} version={version} />
      ));

    return (
      <>
        <ActiveVersionStatus
          activeVersion={activeVersion}
          onClick={onLocateActiveVersionClick}
        />
        <div className={styles.versionList} ref={versionListRef}>
          {versions}
        </div>
      </>
    );
  }

  return (
    <div className={styles.content}>
      <h2>Versions</h2>
      <p className={styles.subtitle}>
        Fusce vehicula dolor arcu, sit amet blandit dolor mollis nec. Donec
        viverra eleifend lacus, vitae ullamcorper metus. Sed sollicitudin ipsum
        quis nunc sollicitudin ultrices. Donec euismod scelerisque ligula.
        Maecenas eu varius risus, eu aliquet arcu. Curabitur fermentum suscipit
        est, tincidunt.
      </p>
      {getContent()}
    </div>
  );
}

export default RuntimeVersions;
