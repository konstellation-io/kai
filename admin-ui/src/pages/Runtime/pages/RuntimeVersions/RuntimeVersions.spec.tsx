import React from 'react';
import { shallow } from 'enzyme';
import { runtime, version } from '../../../../mocks/version';
import RuntimeVersions from './RuntimeVersions';

describe('RuntimeVersions', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <RuntimeVersions runtime={runtime} versions={[version, version]} />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
