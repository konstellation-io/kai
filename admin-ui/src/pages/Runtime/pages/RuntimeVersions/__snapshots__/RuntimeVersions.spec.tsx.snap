// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuntimeVersions matches snapshot 1`] = `
<div
  className="content"
>
  <h2>
    Versions of runtime 
    runtimeName
  </h2>
  <p
    className="subtitle"
  >
    Here you can see all the versions from the opened runtime. Click on any version to open it or add a new version by clicking on the 'ADD NEW VERSION' button
  </p>
  <PublishedVersionStatus
    nPublishedVersions={0}
    noVersions={false}
  />
  <div
    className="versionList"
  >
    <VersionInfo
      key="versionId"
      version={
        Object {
          "__typename": "Version",
          "configurationCompleted": true,
          "creationAuthor": Object {
            "__typename": "User",
            "email": "user@domain.com",
            "id": "userId",
          },
          "creationDate": "2020/01/01",
          "description": "versionDescription",
          "id": "versionId",
          "name": "versionName",
          "publicationAuthor": Object {
            "__typename": "User",
            "email": "user@domain.com",
            "id": "userId",
          },
          "publicationDate": "2020/01/01",
          "status": "STARTED",
        }
      }
    />
    <VersionInfo
      key="versionId"
      version={
        Object {
          "__typename": "Version",
          "configurationCompleted": true,
          "creationAuthor": Object {
            "__typename": "User",
            "email": "user@domain.com",
            "id": "userId",
          },
          "creationDate": "2020/01/01",
          "description": "versionDescription",
          "id": "versionId",
          "name": "versionName",
          "publicationAuthor": Object {
            "__typename": "User",
            "email": "user@domain.com",
            "id": "userId",
          },
          "publicationDate": "2020/01/01",
          "status": "STARTED",
        }
      }
    />
  </div>
</div>
`;
